lib/matplotlib/colors.py

```diff
@@ -1225,6 +1254,875 @@ def reversed(self, name=None):
+        self.name = name
+
+        if not np.iterable(colormaps) \
+           or len(colormaps) == 1 \
+           or isinstance(colormaps, str):
+            raise ValueError("A MultivarColormap must have more than one colormap.")
+        colormaps = list(colormaps)  # ensure cmaps is a list, i.e. not a tuple
+        for i, cmap in enumerate(colormaps):
+            if isinstance(cmap, str):
+                colormaps[i] = mpl.colormaps[cmap]
+            elif not isinstance(cmap, Colormap):
+                raise ValueError("colormaps must be a list of objects that subclass"
+                                 " Colormap or valid strings.")
+
+        self._colormaps = colormaps
+        if combination_mode not in ['sRGB_add', 'sRGB_sub']:
+            raise ValueError("Combination_mode must be 'sRGB_add' or 'sRGB_sub',"
+                             f" {combination_mode!r} is not allowed.")
+        self._combination_mode = combination_mode
+        self.n_variates = len(colormaps)
+        self._rgba_bad = (0.0, 0.0, 0.0, 0.0)  # If bad, don't paint anything.
+
+    def __call__(self, X, alpha=None, bytes=False, clip=True):
+        r"""
+        Parameters
+        ----------
+        X : tuple (X0, X1, ...) of length equal to the number of colormaps
+            X0, X1 ...:
+            float or int, `~numpy.ndarray` or scalar
+            The data value(s) to convert to RGBA.
+            For floats, *Xi...* should be in the interval ``[0.0, 1.0]`` to
+            return the RGBA values ``X*100`` percent along the Colormap line.
+            For integers, *Xi...*  should be in the interval ``[0, self[i].N)`` to
+            return RGBA values *indexed* from colormap [i] with index ``Xi``, where
+            self[i] is colormap i.
+        alpha : float or array-like or None
+            Alpha must be a scalar between 0 and 1, a sequence of such
+            floats with shape matching *Xi*, or None.
+        bytes : bool
+            If False (default), the returned RGBA values will be floats in the
+            interval ``[0, 1]`` otherwise they will be `numpy.uint8`\s in the
+            interval ``[0, 255]``.
+        clip : bool
+            If True, clip output to 0 to 1
+
+        Returns
+        -------
+        Tuple of RGBA values if X[0] is scalar, otherwise an array of
+        RGBA values with a shape of ``X.shape + (4, )``.
+        """
+
+        if len(X) != len(self):
+            raise ValueError(
+                f'For the selected colormap the data must have a first dimension '
+                f'{len(self)}, not {len(X)}')
+        rgba, mask_bad = self[0]._get_rgba_and_mask(X[0], bytes=False)
+        rgba = np.asarray(rgba)
+        for c, xx in zip(self[1:], X[1:]):
+            sub_rgba, sub_mask_bad = c._get_rgba_and_mask(xx, bytes=False)
+            sub_rgba = np.asarray(sub_rgba)
+            rgba[..., :3] += sub_rgba[..., :3]  # add colors
+            rgba[..., 3] *= sub_rgba[..., 3]  # multiply alpha
```

Comment:
Is there a reference / standard name for this method?  It makes some sense, but is not normal compositing.

Reply:
This also has the property that if one colormap comes back with `alpha=0`, then everything is 0, where I'm not sure makes sense.  Should the alpha be pre-multiplied?

Reply:
Truth be told, we have to treat alpha in some way, and in most use cases the implementation we choose does not matter. i.e. we can easily change this if you have an alternative expression you would like to use.

There was one use case I could think about: Two colorbars, one that is a color gradient, and another that is an alpha gradient. This implementation supports that use case, but I think it needs a bit more work to make it user-friendly.

> This also has the property that if one colormap comes back with alpha=0, then everything is 0, where I'm not sure makes sense. 

To me it makes sense that if `alpha=0` on one colormap the result should have `alpha=0`. The use case I am thinking of is a categorical colomap, and you want one of the categories to hide the element (i.e. missing data).

>  Should the alpha be pre-multiplied?

I'm not sure what this would imply, can you describe this with an equation?

Reply:
`rgba[..., :3] += sub_rgba[..., :3] * sub_rgba[..., 3]` is what I meant by "pre-multiply".

I am not sure what the right expression is, but would like what ever it is to be well justified.  Everything I found about blending was always 2 layers (not N) so I'm not sure if there an existing generalization available to us?

Reply:
I don't see any reason to do `sub_rgba[..., :3] * sub_rgba[..., 3]`. 
If we do this operation, then both `[.5, .5, .5, 1]` and `[1, 1, 1, .5]` produce the same color, and the former should always be preferred. I.e. this adds no new functionality to the multivariate colorbars. 
To me, alpha has a unique use with regards to composition (i.e. layering different artists in a figure), and if we start mixing alpha and color, we interfere with this use of alpha. 

The use case I have in mind is something like this:

```py
import matplotlib.pyplot as plt
import numpy as np
import matplotlib

vals = np.zeros((256, 4))
vals[:, 3] = np.linspace(1,0, 256)
am = matplotlib.colors.ListedColormap(vals, 'alpha')
RB = plt.get_cmap('RdBu')
cmap = matplotlib.colors.MultivarColormap((RB, am), combination_mode='sRGB_add')

n = 100
x = np.linspace(-1,1, n)[np.newaxis, :]*np.ones((n,n))
y = np.linspace(-1,1, n)[:, np.newaxis]*np.ones((n,n))
r = np.sqrt(x**2+y**2)
a = np.random.random((n,n))*2-1
b = np.random.random((n,n))


fig, ax = plt.subplots()
ax.set_facecolor("k")
ax.contour(x, y, r, levels=10, colors='w', zorder=0)
cs = ax.pcolormesh(x, y, (a, b), cmap=cmap, zorder=2, vmin =(-1, 0), vmax = (1, 1))

fig.colorbar(cs.colorizer[0], ax=ax).set_label('RdBu')
fig.colorbar(cs.colorizer[1], ax=ax).set_label('Alpha')
```
![image](https://github.com/user-attachments/assets/45d6ad39-3010-4977-aa69-5994704b35d4)

Where different variables are shown using a diverging colorbar and lightness (alpha), and by using alpha, additional information can be shown behind the map (contour lines).
While this shows made-up data, I think you could make interesting maps with this. Consider for example a map of population density (alpha), median age (color) overlaid over a map of major highways.